### ISSUE TITLE
[Request] Remove unused config defines

### ISSUE THREAD

**Issue by helmesjo** [libevent/libevent#1329]
Hi!

I'm currently [packaging `libevent` for `build2`](https://github.com/build2-packaging/libevent) and in the process of doing so I noticed that the config files (eg. `event-config.hpp.cmake`) lists _a lot_ of unused defines. It would be good if any unused defines would be cleaned up to avoid confusion for what is actually required/used.

See attached file for a complete list of all _used_ defines (manually searched each define to verify it is actually used in the code).

Thanks!

[LIBEVENT-defines-used.txt](https://github.com/libevent/libevent/files/9444614/LIBEVENT-defines-used.txt)

**Comment by azat** [comment]
Thank you for the report!

Indeed, I'm pretty sure that there are lots of unused.
Can you also share a script that you were using to collect them?
Also maybe you want to submit this cleanup by yourself, via pull request?

**Comment by helmesjo** [comment]
@azat Just some (temp) regex I made while packaging this, but I'm fairly confident these are all the unused (at least I know these are unused):

```
EVENT__DNS_USE_CPU_CLOCK_FOR_ID
EVENT__DNS_USE_GETTIMEOFDAY_FOR_ID
EVENT__DNS_USE_FTIME_FOR_ID
EVENT__HAVE_TAILQFOREACH
EVENT__HAVE_DLFCN_H
EVENT__HAVE_EPOLL_CTL
EVENT__HAVE_EVENT_PORTS
EVENT__HAVE_LIBZ
EVENT__HAVE_MEMORY_H
EVENT__HAVE_OPENSSL
EVENT__HAVE_POLL_H
EVENT__HAVE_PTHREADS
EVENT__HAVE_SIGNAL
EVENT__HAVE_SPLICE
EVENT__HAVE_STDARG_H
EVENT__HAVE_STRUCT_IN6_ADDR_S6_ADDR16
EVENT__HAVE_STRUCT_IN6_ADDR_S6_ADDR32
EVENT__HAVE_SYS_EPOLL_H
EVENT__HAVE_SYS_EVENT_H
EVENT__HAVE_SYS_QUEUE_H
EVENT__HAVE_TAILQFOREACH
EVENT__HAVE_TIMERCMP
EVENT__SIZEOF_SSIZE_T
EVENT__TIME_WITH_SYS_TIME
EVENT__SIZEOF_SOCKLEN_T
EVENT__size_t
```

**Comment by francoisk** [comment]
`_FILE_OFFSET_BITS` is also unused.

Only reference in source code:
https://github.com/libevent/libevent/blob/b19af675c7601a7867f26c33072cda7ea125adb2/test/regress_zlib.c#L74-L76

And there's also this:
https://github.com/libevent/libevent/blob/b19af675c7601a7867f26c33072cda7ea125adb2/CMakeLists.txt#L707-L715

**Comment by azat** [comment]
>A follow-up request related to this would be to replace all #define with #cmakedefine in event-config.h.cmake (or at least be consistent, since atm it is a mix of both and from the looks of it, just using #cmakedefine should be fine).

@helmesjo Yes, this is indeed what should be done. Maybe you can submit a PR?

**Comment by azat** [comment]
>_FILE_OFFSET_BITS is also unused.

@francoisk this is glibc macro - https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html#index-_005fFILE_005fOFFSET_005fBITS

Though it is tricky and glibc specific.

**Comment by francoisk** [comment]
> > _FILE_OFFSET_BITS is also unused.
> 
> @francoisk this is glibc macro - https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html#index-_005fFILE_005fOFFSET_005fBITS
> 
> Though it is tricky and glibc specific.

@azat OK but there's a check for it in evconfig-private.h.cmake:

https://github.com/libevent/libevent/blob/039e8d96a476a93874576d808242157698496f04/evconfig-private.h.cmake#L20-L21
